tests/examples/test_examples.py

```diff
@@ -28,8 +48,15 @@
     (os.path.join('tensorflow', 'tf1'), ['-P', 'steps=10']),
     ('xgboost', ['-P', 'learning_rate=0.3', '-P', 'colsample_bytree=0.8', '-P', 'subsample=0.9'])
 ])
-def test_mlflow_run_example(directory, params):
-    cli_run_list = [os.path.join(EXAMPLES_DIR, directory)] + params
+def test_mlflow_run_example(directory, params, tmpdir):
+    example_dir = os.path.join(EXAMPLES_DIR, directory)
+    tmpdir = os.path.join(tmpdir.strpath, directory)
+
+    shutil.copytree(example_dir, tmpdir)
+    conda_yml_path = find_conda_yaml(tmpdir)
+    replace_mlflow_with_dev_version(conda_yml_path, os.path.abspath('.'))
```

Comment:
Nit: `os.path.abspath('.')` seems to assume that we're invoking this test from the `~/mlflow` root directory. For debugging purposes / manual execution, it may be better to fetch the root directory path as:

```
import mlflow
os.path.dirname(mlflow.__path__[0])
```